src/components/input/input.js

```diff
@@ -1007,15 +1005,8 @@ function ngMessageAnimation($$AnimateRunner, $animateCss, $mdUtil, $log) {
 function showInputMessages(element, done) {
   var animators = [], animator;
   var messages = getMessagesElement(element);
-  var children = messages.children();
-
-  if (messages.length == 0 || children.length == 0) {
-    $log.warn('mdInput messages show animation called on invalid messages element: ', element);
-    done();
-    return;
-  }
```

Comment:
I can understand removing the warning, but this seems to also be removing the `done();` call and the early `return`. I think that we still want those things for performance and correctness. We don't need to trigger the animations if there are no messages to animate in.

Reply:
This makes sense, thanks. I will implement those changes.

As the `animators` array is empty it should not cause any performance issues. However for the correctness it makes sense to handle it here rather than depend on the behaviour of the called `all` function.

I will fix it as soon as possible.